cdap-ui/app/cdap/components/NamespaceDetails/store/ActionCreator.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import {MyNamespaceApi} from 'api/namespace';
+import {MyPreferenceApi} from 'api/preference';
+import {MySearchApi} from 'api/search';
+import {MyAppApi} from 'api/app';
+import {getCurrentNamespace} from 'services/NamespaceStore';
+import NamespaceDetailsStore, {NamespaceDetailsActions} from 'components/NamespaceDetails/store';
+import {Observable} from 'rxjs/Observable';
+import {GLOBALS} from 'services/global-constants';
+import EntityType from 'services/metadata-parser/EntityType';
+
+function enableLoading() {
+  NamespaceDetailsStore.dispatch({
+    type: NamespaceDetailsActions.enableLoading
+  });
+}
+
+function getNamespacePrefs() {
+  let namespace = getCurrentNamespace();
+
+  MyPreferenceApi
+    .getNamespacePreferences({namespace})
+    .subscribe(
+      (res) => {
+        NamespaceDetailsStore.dispatch({
+          type: NamespaceDetailsActions.setData,
+          payload: {
+            namespacePrefs: res
+          }
+        });
+      },
+      (err) => console.log(err)
+    );
+}
+
+function getData() {
+  enableLoading();
+
+  let namespace = getCurrentNamespace();
+
+  let searchParams = {
+    namespace,
+    target: ['dataset', 'stream'],
+    query: '*'
+  };
+
+  Observable
+    .forkJoin(
+      MyNamespaceApi.get({namespace}),
+      MyPreferenceApi.getNamespacePreferences({namespace}),
+      MyPreferenceApi.getNamespacePreferencesResolved({namespace}),
+      MyAppApi.list({namespace}),
+      MySearchApi.search(searchParams)
+    )
+    .subscribe(
+      (res) => {
+        let {name, description, config} = res[0];
+        let namespacePrefs = {...res[1]};
+        let resolvedPrefs = {...res[2]};
```

Comment:
You don't have to change it but you could do something like this,
```
let [ namespaceInfo, namespacePrefs, resolvedPrefs, pipelines, datasets ] = res;
```

Its a simple spread operator.